labs/b9.md

```diff
@@ -0,0 +1,527 @@
+---
+title: Lab 9 - Version Control and Backups
+layout: lab
+---
+
+## Overview
+
+In this lab, we'll be talking about using version control, in particular Git, for keeping track of changes to your files and backing up your files to other machines. 
+We'll be setting up [Gitolite](http://gitolite.com/gitolite/index.html), an open-source Git server that allows you to host Git repositories on other machines,
+and set access to Git repositories hosted on the Gitolite server.
+
+## A Git Primer
+
+Version control tools are designed to keep track of changes to files, and allow you to rewind to some last saved state. 
+Git, as one of the most popular version control tools in use today, supports these features, but also allows for more advanced workflows such as splitting a filebase into "branches" for experimentation.
+In order to complete this lab, you'll need to know a few basic git commands and some terminology.
+
+### Getting started in git: Initializing/cloning a repository
+
+A "repository" is just a fancy way of saying "a folder where git is set up."
+In order to start working with git, you'll need to either create a repository from scratch,
+or "clone" (download and copy) a repository from somewhere else.
+
+A folder can be made into a git **repository** by running `git init`, telling git that files under the folder can start being tracked by git. When a repository is initialized, git creates a hidden `.git` folder within the root folder to track the files' histories.
+
+```
+bzh@supernova ~> mkdir example-project
+bzh@supernova ~> cd example-project/
+bzh@supernova ~/example-project> git init
+Initialized empty Git repository in /home/b/bz/bzh/example-project/.git/
+```
+
+A git repository can also be **cloned** from online by running `git clone [repository url]`, which will create a copy of the repository
+on your machine, and set the address you cloned the repo from to be the "origin" remote (more on this later).
+
+```
+bzh@supernova /tmp> git clone https://github.com/ocf/ocfweb.git
+Cloning into 'ocfweb'...
+remote: Enumerating objects: 10, done.
+remote: Counting objects: 100% (10/10), done.
+remote: Compressing objects: 100% (10/10), done.
+remote: Total 13063 (delta 1), reused 0 (delta 0), pack-reused 13053
+Receiving objects: 100% (13063/13063), 3.75 MiB | 0 bytes/s, done.
+Resolving deltas: 100% (9159/9159), done.
+
+bzh@supernova /tmp> cd ocfweb
+
+bzh@supernova /t/ocfweb> git status
+On branch master
+Your branch is up-to-date with 'origin/master'.
+nothing to commit, working tree clean
+
+bzh@supernova /t/ocfweb> git remote -v
```

Comment:
likewise, the shorthand /t/ocfweb is may confuse students trying to pattern-match here, I'd remove it